### PR TITLE
Support running tests against a remote docker daemon

### PR DESCRIPTION
**Pull request by adam-stokes**
## What does this PR do?

This PR adds support for connecting to a remote docker instance for deployment/testing

## Why is it important?

Allows the test framework to utilize a remote deployment so that we can easily test agents locally on the host machines for windows and mac

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the Unit tests (`make unit-test`), and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ]

## How to test this PR locally

The remote system should have ssh passwordless key installed, then point the test framework to the remote docker instance:

```
DOCKER_HOST="ssh://user@192.168.8.4" make ...
```

This is the foundational work to address https://github.com/elastic/observability-robots/issues/427 and https://github.com/elastic/observability-robots/issues/474

### CODE REVIEW

File: internal/deploy/docker_client.go [hunk 1]
```diff
@@ -75,6 +75,7 @@ func buildTarForDeployment(file *os.File) (bytes.Buffer, error) {
 // CopyFileToContainer copies a file to the running container
 func CopyFileToContainer(ctx context.Context, containerName string, srcPath string, parentDir string, isTar bool) error {
 	dockerClient := getDockerClient()
+	defer dockerClient.Close()
```
Comment (mdelapenya) [inline internal/deploy/docker_client.go:78]: 🥇

File: internal/deploy/docker.go [hunk 2]
```diff
@@ -18,11 +19,16 @@ import (
...
+	Context          context.Context
+	ConnectionString string
 }
 
 func newDockerDeploy() Deployment {
-	return &dockerDeploymentManifest{Context: context.Background()}
+	isConnectionString := shell.GetEnv("DOCKER_HOST", "")
+	return &dockerDeploymentManifest{
+		Context:          context.Background(),
+		ConnectionString: isConnectionString,
```
Comment (mdelapenya) [inline internal/deploy/docker.go:30]:
```suggestion
	connectionString := shell.GetEnv("DOCKER_HOST", "")
	return &dockerDeploymentManifest{
		Context:          context.Background(),
		ConnectionString: connectionString,
```

**Comment by cachedout** [comment]
@adam-stokes Is this connected to a currently open issue? If so, could you please link it? Thanks!

**Review by mdelapenya — APPROVED** [review]

**Comment by adam-stokes** [comment]
jenkins run the tests please

File: cli/config/compose/profiles/fleet/docker-compose.yml [hunk 3]
```diff
@@ -32,8 +32,21 @@ services:
...
+      - "SERVER_NAME=kibana"
+      - "SERVER_HOST=0.0.0.0"
+      - "TELEMETRY_ENABLED=false"
+      - "ELASTICSEARCH_HOSTS=http://elasticsearch:9200"
+      - "ELASTICSEARCH_USERNAME=elastic"
+      - "ELASTICSEARCH_PASSWORD=changeme"
+      - "MONITORING_UI_CONTAINER_ELASTICSEARCH_ENABLED=true"
+      - "XPACK_ENCRYPTEDSAVEDOBJECTS_ENCRYPTIONKEY='12345678901234567890123456789012'"
+      - "XPACK_FLEET_ENABLED=true"
+      - "XPACK_FLEET_REGISTRYURL=https://epr-staging.elastic.co"
```
Comment (cachedout) [inline cli/config/compose/profiles/fleet/docker-compose.yml:45]:
```suggestion
      - "XPACK_FLEET_REGISTRY_URL=https://epr-staging.elastic.co"
```
Nit: the spacing is inconsistent here

File: cli/config/compose/profiles/fleet/docker-compose.yml [hunk 4]
```diff
@@ -32,8 +32,21 @@ services:
...
+      - "XPACK_FLEET_AGENTS_ENABLED=true"
+      - "XPACK_FLEET_AGENTS_ELASTICSEARCH_HOST=http://elasticsearch:9200"
```
Comment (cachedout) [inline cli/config/compose/profiles/fleet/docker-compose.yml:47]: L39 has this as plural and this is singular. Should they be aligned?

File: cli/config/compose/profiles/fleet/docker-compose.yml [hunk 5]
```diff
@@ -32,8 +32,21 @@ services:
...
+      - "XPACK_FLEET_AGENTS_FLEET_SERVER_HOSTS=http://fleet-server:8220"
+      - "XPACK_FLEET_AGENTS_TLSCHECKDISABLED=true"
```
Comment (cachedout) [inline cli/config/compose/profiles/fleet/docker-compose.yml:49]:
```suggestion
      - "XPACK_FLEET_AGENTS_TLS_CHECK_DISABLED=true"
```
Nit: inconsistent spacing

Comment (adam-stokes) [reply cli/config/compose/profiles/fleet/docker-compose.yml:49]: According to the kibana documentation those environment variables map directly to the config options set in the kibana config

https://www.elastic.co/guide/en/kibana/current/docker.html#environment-variable-config

This makes me think that the underscores and plural naming should match?

Comment (cachedout) [reply cli/config/compose/profiles/fleet/docker-compose.yml:49]: Oh I didn't even notice that these were actually mapping to the Kibana envs. (It's been a long day.) I think you're correct here. I'll resolve these comments.

**Review by cachedout — APPROVED** [review]

**Pull request merged**